RFC 59 - show studies for which a user is not authorized

dianab0 opened the pull request:
Rfc 59 (https://docs.google.com/document/d/1NkNhBG1sk0ZvCOwk3XtiuRT9a_tKkFCgXC9I3XT-Ye4/edit#heading=h.6c2l76kgw1ta)
Physical studies that are not authorized are grayed out, have a gray info icon with no tooltip and a lock icon at then end. This behavior can be enabled/disabled from portal.properties (by default false). Also the global message that appears when hover over lock icon can be configured in portal.properties as well.

![Screenshot from 2021-08-17 12-24-03](https://user-images.githubusercontent.com/53996876/129889271-08613319-a3af-4ca5-a431-f8bb51549b34.png)


Backend PR is https://github.com/cBioPortal/cbioportal/pull/8847

alisman commented on src/pages/studyView/StudyViewPageStore.ts:
```diff
@@ -4120,7 +4120,14 @@ export class StudyViewPageStore
                 this.studyIds,
                 (acc: CancerStudy[], next) => {
-                    if (everyStudyIdToStudy[next]) {
+                    if (
```
lets put this condition in a helper function

alisman commented on src/pages/studyView/StudyViewPageStore.ts:
```diff
@@ -4120,7 +4120,14 @@ export class StudyViewPageStore
+                            .skin_show_unauthorized_studies ||
+                            (AppConfig.serverConfig
+                                .skin_show_unauthorized_studies &&
+                                everyStudyIdToStudy[next].isAuthorized == true))
```
use ===

alisman replied:
== in javascript means type coercion.  it's just a good practice to use === as a rule

alisman commented on src/shared/components/query/studyList/StudyList.tsx:
```diff
@@ -367,27 +372,37 @@ export default class StudyList extends QueryStoreComponent<
-                                    <a>{content}</a>
-                                </StudyTagsTooltip>
-                            );
+                            if (
```
again, lets wrap in a helper function.

dianab0 commented:
Thank you for review @alisman. I updated the code. I added the helper function in both StudyViewPageStore.ts and StudyList.tsx. However in the StudyList we may remove it the future if we'll implement, as per RFC, the retrieval of info icon data for unauthorized studies too.

dianab0 commented:
@alisman sorry, I forgot to push the commit containing the updates after review. It is pushed now.

alisman commented on src/shared/components/query/studyList/StudyList.tsx:
```diff
@@ -66,6 +66,14 @@ export default class StudyList extends QueryStoreComponent<
         if (this.props.showSelectedStudiesOnly) this._view = this.view;
     }
 
+    isQueriedStudyAuthorized = (study: CancerStudy): boolean => {
```
can we put this in utils so we don't define it twice?
